client/signup/index.node.js

```diff
@@ -3,38 +3,19 @@
 /**
  * Internal dependencies
  */
+import { getLanguage, getLanguageSlugs } from 'lib/i18n-utils';
 
-import { getLanguage } from 'lib/i18n-utils';
+const lang = `:lang(${ getLanguageSlugs().join( '|' ) })?`;
 
 export default function( router ) {
-	router( '/start/:flowName?/:stepName?/:stepSectionName?/:lang?', setUpLocale );
+	router( `/start/:flowName?/:stepName?/:stepSectionName?/${ lang }`, setUpLocale );
```

Comment:
There's a lot of dense information here and the `${}` vs page.js route params was hard for me to read. Initially I'd missed the `?` which moved to the declaration above.

Reply:
Yeah, fair point. This is mostly to align with the existing definition in `client/login`

https://github.com/Automattic/wp-calypso/blob/2899fd4a21f3274be0cc8ce632edbdca8bac167c/client/login/controller.js#L43

which is used quite heavily in the corresponding route defs

https://github.com/Automattic/wp-calypso/blob/2899fd4a21f3274be0cc8ce632edbdca8bac167c/client/login/index.web.js#L33-L37

Should we move the `?` out of the definition? What about the `:lang`? The parens? :smile: (Maybe keep the parens, move the rest out?)

Reply:
I don't have a great suggestion or I would've made it 😅 

Mostly just an observation and note for posterity given that it took me a while to spot the moved `?`.